src/internal/words.ts

```diff
@@ -31,30 +31,55 @@ const WHITESPACE = [
 ] as const;
 
 // @see https://github.com/sindresorhus/type-fest/blob/main/source/internal/characters.d.ts#L33
-const WORD_SEPARATORS = ["-", "_", ...WHITESPACE] as const;
-
-// We are basing our logic on the type definition of SplitWords from type-fest.
-// @see https://github.com/sindresorhus/type-fest/blob/main/source/split-words.d.ts
-const WORD_SPLITTING_RE = new RegExp(
-  [
-    // **on** any word separator, these would be removed from the output.
-    `[${WORD_SEPARATORS.join("")}]+`,
-    // When the text transitions from a non-digit to a digit.
-    String.raw`(?<=\D)(?=\d)`,
-    // When the text transitions from a digit to a non-digit.
-    String.raw`(?<=\d)(?=\D)`,
-    // When the text transitions from a lower case to upper case.
-    String.raw`(?<=[a-z])(?=[A-Z])`,
-    // When the text transitions from 2 upper case letters to a lower case
-    // letter. (one upper case letter is considered part of the word, e.g.
-    // "Dog").
-    String.raw`(?<=[A-Z])(?=[A-Z][a-z])`,
-  ].join("|"),
-  "u",
-);
+const WORD_SEPARATORS = new Set(["-", "_", ...WHITESPACE]);
 
 export const words = <S extends string>(
   data: S,
-): string extends S ? Array<string> : Words<S> =>
+): string extends S ? Array<string> : Words<S> => {
+  const results: Array<string> = [];
+  let word = "";
+
+  const flush = (): void => {
+    if (word.length > 0) {
+      results.push(word);
+      word = "";
+    }
+  };
+
+  for (const character of data) {
+    if (WORD_SEPARATORS.has(character)) {
+      // Separator encountered; flush the current word.
```

Comment:
The comment above also mentions more clearly that the separator would be removed, can you mention that herew?